src/main/java/edu/cornell/kfs/concur/service/impl/ConcurAccountValidationServiceImpl.java

```diff
@@ -2,17 +2,218 @@
+                    validationResult.addMessages(subObjectCodeValidationResult.getMessages());
+                }
+
+                if (projectCodeValidationResult.isNotValid()) {
+                    validationResult.setValid(false);
+                    validationResult.addMessages(projectCodeValidationResult.getMessages());
+                }
+
+                return validationResult;
+            }
+        }
+    }
+
+    public ValidationResult checkRequiredAccountInfo(String chartOfAccountsCode, String accountNumber, String objectCode) {
+        ValidationResult validationResult = new ValidationResult(true, new ArrayList<String>());
+        if (chartOfAccountsCode == null || chartOfAccountsCode.isEmpty()) {
+            validationResult.setValid(false);
+            validationResult.addMessage(ConcurConstants.AccountingStringValidationErrorMessages.ERROR_CHART_OF_ACCTS_REQUIRED);
+            return validationResult;
+        } else if (accountNumber == null || accountNumber.isEmpty()) {
+            validationResult.setValid(false);
+            validationResult.addMessage(ConcurConstants.AccountingStringValidationErrorMessages.ERROR_ACCT_NBR_REQUIRED);
+            return validationResult;
+        } else if (objectCode == null || objectCode.isEmpty()) {
+            validationResult.setValid(false);
+            validationResult.addMessage(ConcurConstants.AccountingStringValidationErrorMessages.ERROR_OBJ_CD_REQUIRED);
+            return validationResult;
+        }
 
         return validationResult;
     }
+
+    public ValidationResult checkAccount(String chartOfAccountsCode, String accountNumber) {
+        ValidationResult validationResult = new ValidationResult(true, new ArrayList<String>());
+
+        Account account = accountService.getByPrimaryId(chartOfAccountsCode, accountNumber);
+
+        if ( account == null || account.toString().isEmpty()) {
+            validationResult.setValid(false);
+            validationResult.addMessage(ConcurConstants.AccountingStringValidationErrorMessages.ERROR_ACCT_DOES_NOT_EXIST);
+        } else if (!account.isActive()) {
+            validationResult.setValid(false);
+            validationResult.addMessage(ConcurConstants.AccountingStringValidationErrorMessages.ERROR_ACCT_INACTIVE);
+        } else if (account.isClosed()) {
+            validationResult.setValid(false);
+            validationResult.addMessage(ConcurConstants.AccountingStringValidationErrorMessages.ERROR_ACCT_CLOSED);
+        }
+        return validationResult;
+    }
+
+    public ValidationResult checkObjectCode(String chartOfAccountsCode, String objectCodeParm) {
+        ValidationResult validationResult = new ValidationResult(true, new ArrayList<String>());
+        
+        ObjectCode objectCode = objectCodeService.getByPrimaryIdForCurrentYear(chartOfAccountsCode, objectCodeParm);
+    
+        if (objectCode == null || objectCode.toString().isEmpty()) {
```

Comment:
I don't know if this would make the code cleaner or not, but this block is pretty similar between most of the accounting string components. I think you could extract it into a method with three parameters (Inactivatable object, String missingMessage, String inactiveMessage) with common logic for all of them.